src/ngraph/frontend/onnx_import/op/lstm.cpp

```diff
@@ -55,23 +64,63 @@ namespace ngraph
+                                                  const std::shared_ptr<ngraph::Node>& rhs)
+                {
+                    auto args = numpy_style_broadcast_for_binary_operation(lhs, rhs);
+                    return {std::make_shared<ngraph::op::Subtract>(args.at(0), args.at(1))};
+                }
+
                 std::shared_ptr<ngraph::Node> mul(const std::shared_ptr<ngraph::Node>& lhs,
                                                   const std::shared_ptr<ngraph::Node>& rhs)
                 {
                     auto args = numpy_style_broadcast_for_binary_operation(lhs, rhs);
                     return {std::make_shared<ngraph::op::Multiply>(args.at(0), args.at(1))};
                 }
 
-                // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ACTIVATION FUNCTIONS ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+                std::shared_ptr<ngraph::Node> clip(const std::shared_ptr<ngraph::Node>& data,
+                                                   float threshold)
+                {
+                    if (threshold == 0.f)
+                    {
+                        return data;
+                    }
+
+                    float min_val = -threshold;
+                    float max_val = threshold;
+                    std::size_t size = ngraph::shape_size(data->get_shape());
+                    const std::shared_ptr<ngraph::Node> min_val_node =
+                        ngraph::op::Constant::create(data->get_element_type(),
+                                                     data->get_shape(),
+                                                     std::vector<float>(size, min_val));
+                    const std::shared_ptr<ngraph::Node> max_val_node =
+                        ngraph::op::Constant::create(data->get_element_type(),
+                                                     data->get_shape(),
+                                                     std::vector<float>(size, max_val));
+
+                    return std::make_shared<ngraph::op::Minimum>(
+                        max_val_node, std::make_shared<ngraph::op::Maximum>(data, min_val_node));
+                }
+
+                std::string& to_lower_case(std::string&& s)
+                {
+                    std::transform(std::begin(s), std::end(s), std::begin(s), ::tolower);
+                    return s;
+                }
 
-                std::shared_ptr<ngraph::Node> sigmoid(const std::shared_ptr<ngraph::Node>& arg)
+                std::string& to_lower_case(std::string& s)
                 {
-                    return std::make_shared<ngraph::op::Sigmoid>(arg);
+                    std::transform(std::begin(s), std::end(s), std::begin(s), ::tolower);
+                    return s;
                 }
 
-                std::shared_ptr<ngraph::Node> tanh(const std::shared_ptr<ngraph::Node>& arg)
+                std::vector<std::string>& to_lower_case(std::vector<std::string>&& vs)
```

Comment:
If you are modifying the source vector then why return a vector? You should add a comment about the fact that this modifies the source vector.

Reply:
My intent was:
* to modify input vector (which will be a returned value from other function) in-place
* return a reference to modified vector, since it will be used as initializer for struct member in constructor initializer list. See here:
https://github.com/NervanaSystems/ngraph/blob/985c7d6f4713263bde98293bc618c59891c56292/src/ngraph/frontend/onnx_import/op/lstm.cpp#L259-L261
* Avoid copying of modified vector.